internal/sdkv2provider/resource_cloudflare_zone_cache_reserve.go

```diff
@@ -0,0 +1,138 @@
+package sdkv2provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cacheReserveEnabled  = "on"
+	cacheReserveDisabled = "off"
+)
+
+func resourceCloudflareZoneCacheReserve() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareZoneCacheReserveSchema(),
+		CreateContext: resourceCloudflareZoneCacheReserveCreate,
+		ReadContext:   resourceCloudflareZoneCacheReserveRead,
+		UpdateContext: resourceCloudflareZoneCacheReserveUpdate,
+		DeleteContext: resourceCloudflareZoneCacheReserveDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareZoneCacheReserveImport,
+		},
+		Description: heredoc.Doc(`
+			Provides a Cloudflare Cache Reserve resource. Cache Reserve can
+			increase cache lifetimes by automatically storing all cacheable
+			files in Cloudflare's persistent object storage buckets.
+
+			Note: Using Cache Reserve without Tiered Cache is not recommended.
+
+			Requires Cache Reserve subscription.
```

Comment:
we can omit this as we don't use treat the terraform docs as source of what subscriptions are required for a product to work.

```suggestion
```

Reply:
Got it. Will fix. My reasoning was to ensure that users are aware that there are some requirements for this to work - otherwise, they will get a cryptic error message back, as I did. :smile:

Reply:
Done. Thank you!